consensus/dummy/dynamic_fee_window_test.go

```diff
@@ -0,0 +1,246 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package dummy
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/ava-labs/coreth/params"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDynamicFeeWindow_Add(t *testing.T) {
+	tests := []struct {
+		name     string
+		window   DynamicFeeWindow
+		amount   uint64
+		expected DynamicFeeWindow
+	}{
+		{
+			name: "normal_addition",
+			window: DynamicFeeWindow{
+				1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+			},
+			amount: 5,
+			expected: DynamicFeeWindow{
+				1, 2, 3, 4, 5, 6, 7, 8, 9, 15,
+			},
+		},
+		{
+			name: "amount_overflow",
+			window: DynamicFeeWindow{
+				1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+			},
+			amount: math.MaxUint64,
+			expected: DynamicFeeWindow{
+				1, 2, 3, 4, 5, 6, 7, 8, 9, math.MaxUint64,
+			},
+		},
+		{
+			name: "window_overflow",
+			window: DynamicFeeWindow{
+				1, 2, 3, 4, 5, 6, 7, 8, 9, math.MaxUint64,
+			},
+			amount: 5,
+			expected: DynamicFeeWindow{
+				1, 2, 3, 4, 5, 6, 7, 8, 9, math.MaxUint64,
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			test.window.Add(test.amount)
+			require.Equal(t, test.expected, test.window)
+		})
+	}
+}
+
+func TestDynamicFeeWindow_Shift(t *testing.T) {
+	window := DynamicFeeWindow{
+		1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+	}
+	tests := []struct {
+		amount   uint64
+		expected DynamicFeeWindow
+	}{
+		{
+			amount:   0,
+			expected: window,
+		},
+		{
+			amount: 1,
+			expected: DynamicFeeWindow{
+				2, 3, 4, 5, 6, 7, 8, 9, 10,
+			},
+		},
+		{
+			amount: 2,
+			expected: DynamicFeeWindow{
+				3, 4, 5, 6, 7, 8, 9, 10,
+			},
+		},
+		{
+			amount: 3,
+			expected: DynamicFeeWindow{
+				4, 5, 6, 7, 8, 9, 10,
+			},
+		},
+		{
+			amount: 4,
+			expected: DynamicFeeWindow{
+				5, 6, 7, 8, 9, 10,
+			},
+		},
+		{
+			amount: 5,
+			expected: DynamicFeeWindow{
+				6, 7, 8, 9, 10,
+			},
+		},
+		{
+			amount: 6,
+			expected: DynamicFeeWindow{
+				7, 8, 9, 10,
+			},
+		},
+		{
+			amount: 7,
+			expected: DynamicFeeWindow{
+				8, 9, 10,
+			},
+		},
+		{
+			amount: 8,
+			expected: DynamicFeeWindow{
+				9, 10,
+			},
+		},
+		{
+			amount: 9,
+			expected: DynamicFeeWindow{
+				10,
+			},
+		},
+		{
+			amount:   params.RollupWindow,
```

Comment:
I'd use a raw 10 here (or as another test). What's important is that this is shifting by one more than the last test (i.e. the edge case), not that it's by the actual value of `params.RollupWindow`, even though they're the same thing.